### PR TITLE
Second infrastructure PR

### PR DESCRIPTION
**Pull request by snim2**
This PR allows the user to render JSON output. Format matches GumTree output. Default is to produce no output.

e.g. 

```sh
>>> ./target/debug/rstreediff -m Myers -o Json tests/empty.calc tests/one.calc
{
    "matches": [
        {
            "src": 2,
            "dest": 2
        },
        {
            "src": 1,
            "dest": 1
        },
        {
            "src": 3,
            "dest": 3
        },
        {
            "src": 0,
            "dest": 0
        }
    ],
    "actions": [
        {
            "action": "insert",
            "tree": 2,
            "parent": 2,
            "at": 0
        }
    ]
}
```

The output matches GumTree output and is valid JSON:
```sh
>>> ./target/debug/rstreediff -m Myers -o Json tests/Test0.java tests/Test1.java >output.json

>>> import json
>>> with open("output.json") as fp:
...     j = json.loads(fp.read())
...
>>> j['matches']
[{u'dest': 2, u'src': 2}, {u'dest': 1, u'src': 1}, {u'dest': 6, u'src': 12}, {u'dest': 9, u'src': 5}, {u'dest': 10, u'src': 6}, {u'dest': 12, u'src': 8}, {u'dest': 3, u'src': 11}, {u'dest': 8, u'src': 4}, {u'dest': 7, u'src': 3}, {u'dest': 5, u'src': 10}, {u'dest': 11, u'src': 7}, {u'dest': 4, u'src': 9}, {u'dest': 0, u'src': 0}]

>>> len(j['matches'])
13

>>> len(j['actions'])
5
>>>

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -101,6 +103,12 @@ enum Matchers {
     Myers,
 }
 
+#[derive(Debug, Eq, PartialEq, RustcDecodable)]
+enum Output {
+    NoOutput, // Default.
+    Json,
```
Comment (ltratt) [inline src/main.rs]: Should this be `JSON` (all caps)?

**Comment by ltratt** [comment]
I only have 1 (very, very trivial) comment.

File: src/main.rs (hunk 1 above)
Comment (snim2) [inline src/main.rs]: Good question, I can never figure out the Rust conventions, but I think this *can* be all caps, in which case the other enums in main should be too.

Comment (ltratt) [inline src/main.rs]: I don't think the others should be: it's just that JSON really is an acronym in normal day-to-day life.

**Comment by snim2** [comment]
OK, good to squash?

**Comment by ltratt** [comment]
Yep!

**Comment by snim2** [comment]
Done

**Pull request merged**